app/src/main/java/com/example/myapplication/fragments/MyContactsFragment.kt

```diff
@@ -0,0 +1,117 @@
+import com.example.myapplication.databinding.FragmentMyContactsBinding
+import com.example.myapplication.model.User
+import com.example.myapplication.viewModel.UsersListViewModel
+import com.example.myapplication.viewModel.ViewModelFactory
+import com.google.android.material.snackbar.Snackbar
+
+
+class MyContactsFragment : Fragment() {
+
+    private var _binding: FragmentMyContactsBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var adapter: UsersAdapter
+    private lateinit var viewModel: UsersListViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentMyContactsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val viewModelFactory = ViewModelFactory(requireContext().applicationContext as App)
+        viewModel = ViewModelProvider(this, viewModelFactory)[UsersListViewModel::class.java]
+
+        adapter = UsersAdapter(object : UserActionListener {
+            override fun onDeleteUser(user: User) {
+                val listBeforeDeletedContact = viewModel.usersLiveData.value
+                viewModel.deleteUser(user)
+
+                Snackbar.make(
+                    requireView(),
+                    getString(R.string.contact_has_been_removed),
+                    Snackbar.LENGTH_LONG
+                ).setAction(
+                    getString(R.string.cancel)
+                ) {
+                    viewModel.restoreUser(listBeforeDeletedContact)
+                }
+                    .setActionTextColor(
+                        ContextCompat.getColor(
+                            requireContext(),
+                            R.color.my_light_primary
+                        )
+                    )
+                    .show()
+            }
+
+            override fun onUserDetails(user: User) {
+                val action =
+                    MyContactsFragmentDirections.actionMyContactsFragmentToContactsProfileFragment(
+                        user.photo,
+                        user.name,
+                        user.career
+                    )
+                findNavController().navigate(action)
+            }
+        })
+
+        val layoutManager = LinearLayoutManager(requireContext())
+        binding.recyclerView.layoutManager = layoutManager
+        binding.recyclerView.adapter = adapter
+        viewModel.usersLiveData.observe(viewLifecycleOwner) {
+            adapter.submitList(it.toMutableList())
+        }
+
+        binding.addContactTextView.setOnClickListener {
+            showAddUserDialog()
+        }
+        setupDialogFragmentListener()
+
+        binding.arrowBackImageView.setOnClickListener {
+            findNavController().popBackStack()
+        }
```

Comment:
Better divide observers and clicklisteners by logic groups. For example,  observers in one function and clickListeners in other function.